### ISSUE TITLE
MSSQL Extension will not run

### ISSUE THREAD

**Issue by afittro** [Microsoft/vscode-mssql#711]
- MSSQL Extension Version: 0.2.1
- VSCode Version: 1.9.0 (Date 2017-02-02T08:31:00.827Z); Shell 1.4.6; Node 6.5.0
- OS Version: Windows 7 pro x64, SP1

Steps to Reproduce:

1. Install extension through the UI
2. Open any .sql file
3. Error: "mssql: The platform is not supported"
-or-
1. Install extension through the UI
2. Restart VSCode
3. F1->sqlman->{enter}
4. Errors:
"Running the contributed command:'extension.manageProfiles' failed." and
"mssql: Unable to execute the command while the extension is initializing. Please try again later."

Attempts to fix:
1. Uninstall + Reinstall many times; with restarts of VSCode in each case.
2. Various google searches with suggestions to install sqltools, or a zip of MS content to "%USERPROFILE%\.vscode\extensions\ms-mssql.mssql-0.2.1\sqltoolsservice\0.1.2\Windows"

pics of all MS items installed:

![pic2](https://cloud.githubusercontent.com/assets/541872/23193817/a7315ce4-f85f-11e6-8214-3762eb42133f.png)
![pic3](https://cloud.githubusercontent.com/assets/541872/23193815/a72cdc64-f85f-11e6-96cd-7acec298564a.png)
![pic1](https://cloud.githubusercontent.com/assets/541872/23193816/a72e478e-f85f-11e6-934e-af67b55a119b.png)

Update 2/22/17 - I am also trying to resolve by uninstalling MSSQL extension, upgrading VSCode to 1.9.1 and reinstalling the extension. It's been in "installing" status for 30m with no joy yet...

**Comment by kburtram** [comment]
@afittro It sounds like there is some issue downloading the tools service binaries.  Do you know if there is a proxy or other special network configuration in your environment.

You can try to manually download and install the tools service binaries to see if that works-around the issue.  To do this please download the ZIP file from https://download.microsoft.com/download/1/2/9/129DBE92-1B54-4024-9399-CE31DF148BD8/microsoft.sqltools.servicelayer-win-x64-netcoreapp1.0.zip and unzip the contents directly into C:\Users\[YOUR_USER_NAME]\.vscode\extensions\ms-mssql.mssql-0.2.1\sqltoolsservice\0.2.0\Windows

**Comment by afittro** [comment]
I did several things to try to resolve since last posting. I ripped out the no-longer-licensed VS2015, web platform installer and a few other smaller bits (not related to runtimes and such and guaranteed not to affect this sort of work.) No joy in all 3 attempts. After each attempt I did the typical heavy-handed clean slate process: rip out mssql, reboot vscode and also ensured the extension's directory had no mssql folder.

* I set "http.proxyStrictSSL": false

* I then tried to install the alpha 3.x version from their github

* I then tried to install the netcoreapp1.0. I had done this before, but I thought I'd give it another go. Maybe I had a typo or whatever last time.

![pic1](https://cloud.githubusercontent.com/assets/541872/23285684/a2636a16-f9e7-11e6-8a31-de35c05c602f.png)

![pic2](https://cloud.githubusercontent.com/assets/541872/23285732/deafd3b0-f9e7-11e6-90da-871f85d582da.png)

**Comment by kevcunnane** [comment]
@afittro could you try out the pre-packaged version with offline installer for the service? Our RC for the next version is at https://github.com/Microsoft/vscode-mssql/releases/download/v0.3.0/mssql-0.3.0-win7-x64.vsix. This will definitely have the full required content, so if something goes wrong it's a bug we can try to capture rather than a proxy issue or something else.

**Comment by afittro** [comment]
Thank you all for helping, btw. I did ask requested by @kevcunnane and no joy.

I ripped out any existing mssql install, checking the extensions folder to ensure all was gone.

I downloaded from the link provided, did the install from vsix operation and vscode said the extension installed successfully, click here to reload. Doing both that reload (without closing vscode) and then retrying by rebooting vscode both had the same effects.

1. The extension still gives the 3 identical errors I posted above.
2. The Extensions pane shows it as v 0.2.1
1 pic shows it announcing the wrong version in vscode, the next pic shows the extensions folder with the proper install of 0.3.0 (for the extension itself - see the path at the top) and the sqltoolsservice (in the content area of the folder view below).

![pic1](https://cloud.githubusercontent.com/assets/541872/23287513/c3f7e2f0-f9f2-11e6-96c1-5bb38ef91414.png)

![pic2](https://cloud.githubusercontent.com/assets/541872/23287514/c84a3ede-f9f2-11e6-9879-ddf8789d838f.png)

**Comment by kevcunnane** [comment]
And reloading VSCode doesn't fix this? It looks like there's some corruption going on if it shows both Reload and Install. For reference, here's how it should look after installing that update:
![image](https://cloud.githubusercontent.com/assets/10819925/23287852/c4cc8fee-f9f4-11e6-9029-d57c0e52dfae.png)

As you can see, it's listed as 0.3.0. I am struggling to see what's going wrong here without you opening the Developer Tools in VSCode and debugging / looking for errors in the Console.

**Comment by afittro** [comment]
Yes indeed, I did restart vscode. It's all listed in my comment. To reiterate, I downloaded from the link provided in your prior post, did Install from VSIX, vscode reported the extension installed properly, I rebooted vscode and checked the Extensions pane and it showed 0.2.1, proven by the screenshot I made at that moment.

I've no fear of opening dev tools if anyone thinks they can get to the bottom of this through that path. Absent that, I'll retry from scratch ASAP. I will unstall vscode, remove any references to it in the file system and registry, download the latest from MS and install only the mssql extension 0.3.0 as listed above and report back here.

**Comment by afittro** [comment]
@kevcunnane the url https://github.com/Microsoft/vscode-mssql/releases/download/v0.3.0/mssql-0.3.0-win7-x64.vsix no longer is valid. I see you're up to 0.3.0.2 but cannot figure out the right url to dl the latest version of the vsix file

**Comment by afittro** [comment]
OK, I've reinstalled **everything** from scratch and still have the problems. Details below.

1. I uninstalled vscode 1.9.1 - thoroughly. Uninstaller ran without issues. I then ensured Program Files (x86) had no mention. I checked AppData (local-locallow-roaming) to ensure no mention of vscode. I removed the .vscode folder in my user profile. I searched for references to vscode in the registry. The thing was truly gone.

2. I installed vscode 1.9.1 from the downloaded .exe file from MS. No issues.

3. I ran vscode once. No extensions were installed at this time.

4. I downloaded the final 0.3.0 vsix file https://github.com/Microsoft/vscode-mssql/releases/tag/v0.3.0.2/mssql-0.3.0-win7-x64.vsix via the usual procedure for installing vsix's. The installation went well (pics below) and there was no confusion on version # as before. (ps I think this was caused by a mention of 0.2.1 under %userprofile%.vscode - the extensions folder listed 0.2.1 there even after uninstalling in one of my prior tests. It was there when I was ripping out items in step 1 above.)

5. I restarted vscode and saw that 0.3.0 was indeed installed.

6. I typed f1->sqlman and got the trio of errors. I have included a pic of the install folder. FYI it has version 0.3.0.1 of sqltoolsservice as it should be.

![vscode-mssql-030-install 1](https://cloud.githubusercontent.com/assets/541872/23377324/844533ac-fce4-11e6-925d-1affee3d3c75.png)

![vscode-mssql-030-install 2](https://cloud.githubusercontent.com/assets/541872/23377334/88955c16-fce4-11e6-8dc6-b4333b7c61bd.png)

![vscode-mssql-030-install 3](https://cloud.githubusercontent.com/assets/541872/23377340/8b57286c-fce4-11e6-91b9-e1c958e64f37.png)

**Comment by kevcunnane** [comment]
@afittro - glad you found the build OK. We're learning from this, and one thing is I won't be deleting any tags in the future to avoid the problem you had in testing.

Regarding the bug - I would recommend opening the developer tools, and going to the Console section to look for detailed diagnostics. We should see any Javascript errors you find there. Let us know what you see.


![image](https://cloud.githubusercontent.com/assets/10819925/23377507/1c22a5a6-fce5-11e6-863d-05ba655d92ee.png)

![image](https://cloud.githubusercontent.com/assets/10819925/23377653/a898425c-fce5-11e6-89f0-ab2b7e5a0dde.png)

**Comment by afittro** [comment]
This is what I get when I save the log. Since it doesn't seem to show all that is visible in the log, I thought I'd include screenshots with the various sections expanded. Thank you.

[Extension Host] Unhandled Exception occurred. error: Invalid Platform method: initialize
mssql: The platform is not supported
Activating extension `ms-mssql.mssql` failed: {1}.
[Extension Host] Activating extension `ms-mssql.mssql` failed: {1}.
[Extension Host] Activating extension `ms-mssql.mssql` failed:  undefined
mssql: Unable to execute the command while the extension is initializing. Please try again later.
Running the contributed command:'extension.manageProfiles' failed.
[Extension Host] Here is the error stack:  undefined
Running the contributed command:'extension.manageProfiles' failed.
WARNING: Promise with no error callback:81
Object {exception: null, error: "Running the contributed command:'extension.manageProfiles' failed.", promise: n.C…s.d…e._oncancel, handler: undefined, id: 81…}error: "Running the contributed command:'extension.manageProfiles' failed."exception: nullhandler: undefinedid: 81parent: undefinedpromise: n.Class.derive._oncancel_errorId: 81_isException: false_listeners: null_nextState: null_oncancel: null_state: Object_value: "Running the contributed command:'extension.manageProfiles' failed."done: (e,t)then: (e,t)__proto__: e__proto__: Object__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Object()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()propertyIsEnumerable: propertyIsEnumerable()toLocaleString: toLocaleString()toString: toString()valueOf: valueOf()get __proto__: __proto__()set __proto__: __proto__()
Cannot read property 'onDidOpenTextDocument' of undefined: TypeError: Cannot read property 'onDidOpenTextDocument' of undefined
    at MainController.onDidOpenTextDocument (C:\Users\afittro\.vscode\extensions\ms-mssql.mssql-0.3.0\out\src\controllers\mainController.js:404:28)
    at _vscodeWrapper.onDidOpenTextDocument.params (C:\Users\afittro\.vscode\extensions\ms-mssql.mssql-0.3.0\out\src\controllers\mainController.js:93:66)
    at e.invoke (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:1748)
    at e.fire (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:5235)
    at t.$acceptModelAdd (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:12:31892)
    at t.e.handle (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:12:2729)
    at s (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:8:15667)
    at d (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:8:16344)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

![pic1](https://cloud.githubusercontent.com/assets/541872/23416421/a4b4d3aa-fd97-11e6-9ddd-26f804bd5b0d.png)

![pic2](https://cloud.githubusercontent.com/assets/541872/23416437/b0cac0c8-fd97-11e6-9882-9815e910d6fa.png)

![pic3](https://cloud.githubusercontent.com/assets/541872/23416449/b495f8e4-fd97-11e6-9c1e-72a1fdf98fce.png)

![pic4](https://cloud.githubusercontent.com/assets/541872/23416453/b7e4ca84-fd97-11e6-85f9-0227a6fc4dbe.png)

![pic5](https://cloud.githubusercontent.com/assets/541872/23416458/ba48fd36-fd97-11e6-8ad5-abb1aa7dc223.png)

**Comment by kevin100702** [comment]
extensionHost.ts:282[Extension Host] 激活扩展“ms-mssql.mssql”失败: {1}。d.logExtensionHostMessage @ extensionHost.ts:282
messageService.ts:126mssql: The platform is not supportede.doShow @ messageService.ts:126
mainThreadExtensionService.ts:110激活扩展“ms-mssql.mssql”失败: {1}。t.$localShowMessage @ mainThreadExtensionService.ts:110
extensionHost.ts:282[Extension Host] Activating extension `ms-mssql.mssql` failed:  undefinedd.logExtensionHostMessage @ extensionHost.ts:282
extensionHost.ts:282[Extension Host] Here is the error stack:  undefined
shell.ts:437 Running the contributed command:'extension.connect' failed.e.onUnexpectedError @ shell.ts:437(anonymous function) @ shell.ts:389e.onUnexpectedError @ errors.ts:68o @ errors.ts:88(anonymous function) @ winjs.base.js:43(anonymous function) @ winjs.base.js:38
winjs.base.js:45 WARNING: Promise with no error callback:77
winjs.base.js:46 Object {exception: null, error: "Running the contributed command:'extension.connect' failed.", promise: n.C…s.d…e._oncancel, handler: undefined, id: 77…}
messageService.ts:126 Running the contributed command:'extension.connect' failed.e.doShow @ messageService.ts:126e.show @ messageService.ts:105e.resume @ messageService.ts:64(anonymous function) @ workbench.ts:840e.invoke @ callbackList.ts:63e.fire @ event.ts:118(anonymous function) @ quickOpenController.ts:503
messageService.ts:126 mssql: Unable to execute the command while the extension is initializing. Please try again later.e.doShow @ messageService.ts:126e.show @ messageService.ts:105e.resume @ messageService.ts:64(anonymous function) @ workbench.ts:840e.invoke @ callbackList.ts:63e.fire @ event.ts:118(anonymous function) @ quickOpenController.ts:503

**Comment by kevcunnane** [comment]
@afittro thanks, this is very helpful. I think I've found the root issue here, and it's something that the C# extension for VSCode just fixed in their [1.7 release](https://github.com/OmniSharp/omnisharp-vscode/blob/8299f872239a45f224ae26113b1ad2470174a4d7/CHANGELOG.md). 
- We use the same method they did before now to check if the machine is x64 or x86, which is to call `wmic os get osarchitecture` and process the return string.
- In your case, this clearly isn't working, and so we return Unsupported Platform instead of trying for a good default value.
- Their exact fix is "Better detection of Windows architecture (x86 or x64) when determining extension dependencies to download. The detection logic now uses well-known environment variables rather than launching 'wmic'. (#1110, #1125)". We can use this as a base for our fix.

**Request to you**
Can you go to the Output view, choose `SqlToolsService Initialization` and show us the result? It should list the platform information that we got back.
![image](https://cloud.githubusercontent.com/assets/10819925/23449049/6da5e2b0-fe09-11e6-9803-73b55979277d.png)

**Additional issues**
The other error messages happen mostly because we aren't skipping processing events when initialization failed / didn't complete. I'm adding correct handling for these so they shouldn't occur, but it won't fix the root issue that the platform isn't detected right. For that, we need to update our logic.

**Comment by kevcunnane** [comment]
I have PR https://github.com/Microsoft/vscode-mssql/pull/744 out to fix this.

**Comment by afittro** [comment]
Wow. Great work.

`SQLToolsService Initialization` returned:
```
Initializing SQL tools service for the mssql extension.
Note: mssql commands will be available after installing the service.

Platform: win32, unknown
```

**Comment by kevcunnane** [comment]
@afittro the 0.3.0 release we just published *should* fix this. Please try updating and let us know if it's finally fixed for you.

**Comment by afittro** [comment]
I installed the newly released 0.3.0 on both laptop and desktop. The installation is now flawless. Kudos!

The output of `SqlToolsService Initialization` is now `Platform: win32, x86_64 (Windows)`

When I enter F1->sqlman and try to create a profile, the process works fine until I give it a name and Save. At that point I get
`mssql: Error: Unable to load connection profiles from [C:\Users\afittro\AppData\Roaming\Code\User\settings.json]. Check if the file is formatted correctly.`

I get that same error if I later open an SQL file and do F1->sqlcon or any other mssql command. Well, the error is different for MSSQL: Use Database. That one tells me no connection was found. Please connect first.

That file seems to have saved perfectly though.
`// Place your settings in this file to overwrite the default settings
{
    "mssql.logDebugInfo": true,
    "mssql.maxRecentConnections": 5,
    "mssql.connections": [
        {
            "server": "{{dbprod}}",
            "database": "{{tblName}}",
            "user": "{{me}}",
            "password": "{{secret}}"
        }
    ],
    "mssql.format.keywordCasing": "lowercase",
    "editor.tabSize": 2,
    "extensions.autoUpdate": true,
    "http.proxyStrictSSL": false,
}`

Should I create another Issue for this?

**Comment by kevcunnane** [comment]
Glad you're partway there, now we definitely want to figure out what's wrong in that file. We just use a simple comment-json module to parse so it's strange that this is happening.

Please create a different issue for this. Also could you email me the file contents (remove any password / obfuscate the server name without changing formatting) so I can try to reproduce? I'm kcunnane@microsoft.com. I'll work to repro locally, and then unblock you before seeing about a fix. it's the first report we've had about this error, so not sure what's going wrong.

**Comment by afittro** [comment]
Thank you @kevcunnane for the fixes and hard work. I will now create a new issue to resolve this new problem.

Cool fact: VSCode on Linux is flawless :P, including this extension. I haven't tested it as much but no installation problems for mssql or sqltoolsservice or this settings.json issue
